parsl/providers/kubernetes/kube.py

```diff
@@ -161,42 +159,38 @@ def __init__(self,
         self.resources: Dict[object, Dict[str, Any]]
         self.resources = {}
 
-    def submit(self, cmd_string, tasks_per_node, job_name="parsl"):
+    def submit(self, cmd_string, tasks_per_node: int, job_name: str = "parsl.kube"):
         """ Submit a job
         Args:
              - cmd_string  :(String) - Name of the container to initiate
-             - tasks_per_node (int) : command invocations to be launched per node
 
         Kwargs:
              - job_name (String): Name for job
 
         Returns:
              - job_id: (string) Identifier for the job
         """
+        job_id = uuid.uuid4().hex[:8]
 
-        cur_timestamp = str(time.time() * 1000).split(".")[0]
-        job_name = "{0}-{1}".format(job_name, cur_timestamp)
-
-        if not self.pod_name:
-            pod_name = '{}'.format(job_name)
-        else:
-            pod_name = '{}-{}'.format(self.pod_name,
-                                      cur_timestamp)
+        pod_name = self.pod_name or job_name
+        pod_name = sanitize_dns_subdomain_rfc1123(pod_name)
+        pod_name = pod_name[:253 - 1 - len(job_id)].rstrip(".")  # Leave room for the job ID
+        pod_name = f"{pod_name}.{job_id}"
```

Comment:
this mutate after sanitization made me a bit suspicious.

if some user of the provider submits a job name of `""` and has also configured `self.pod_name = ""` then this can result in an invalid pod name, I think.

In current parsl, that won't happen because job_name is always specificed as beginning with `p`, at:

https://github.com/Parsl/parsl/blob/29f960f264c2bb959f65c07faafb07d20402ad5a/parsl/executors/status_handling.py#L249

So it's not a super big deal

Reply:
Good catch; I'll update the last line to handle this scenario:
```python
pod_name = f"{pod_name}.{job_id}" if pod_name else job_id
```

Reply:
I just modified the code to use `f"parsl.kube.{job_id}"` if the pod name is empty after sanitization.